modules/transport/PostMessageTransportBackend.js

```diff
@@ -0,0 +1,163 @@
+import Postis from 'postis';
+
+/**
+ * The default options for postis.
+ *
+ * @type {Object}
+ */
+const DEFAULT_POSTIS_OPTIONS = {
+    window: window.opener || window.parent
+};
+
+/**
+ * The list of methods of incomming postis messages that we have to support for
+ * backward compatability for the users that are directly sending messages to
+ * Jitsi Meet (without using external_api.js)
+ *
+ * @type {string[]}
+ */
+const LEGACY_INCOMING_METHODS = [
+    'avatar-url',
+    'display-name',
+    'email',
+    'toggle-audio',
+    'toggle-chat',
+    'toggle-contact-list',
+    'toggle-film-strip',
+    'toggle-share-screen',
+    'toggle-video',
+    'video-hangup'
+];
+
+/**
+ * The list of methods of outgoing postis messages that we have to support for
+ * backward compatability for the users that are directly listening to the
+ * postis messages send by Jitsi Meet(without using external_api.js).
+ *
+ * @type {string[]}
+ */
+const LEGACY_OUTGOING_METHODS = [
+    'display-name-change',
+    'incoming-message',
+    'outgoing-message',
+    'participant-joined',
+    'participant-left',
+    'video-conference-joined',
+    'video-conference-left',
+    'video-ready-to-close'
+];
+
+/**
+ * The postis method used for all messages.
+ *
+ * @type {string}
+ */
+const POSTIS_METHOD_NAME = 'data';
+
+/**
+ * Implements message transport using the postMessage API.
+ */
+export default class PostMessageTransportBackend {
+    /**
+     * Creates new PostMessageTransportBackend instance.
+     *
+     * @param {Object} options - Optional parameters for configuration of the
+     * transport.
+     */
+    constructor({ enableLegacyFormat, postisOptions } = {}) {
+        this.postis = Postis({
+            ...DEFAULT_POSTIS_OPTIONS,
+            ...postisOptions
+        });
+
+        this._enableLegacyFormat = enableLegacyFormat;
+
+        if (!this._enableLegacyFormat) {
```

Comment:
I'm not sure I understand the intended meaning of `_enableLegacyFormat`. You always negate it when you want to execute legacy format-related functions/methods...

1. Generally, `_enableLegacyFormat` is a field so our coding style requires jsdocs.

2. Please looks at the uses of `_enableLegacyFormat` (and `enableLegacyFormat`) to ensure that the source code implements the intended functionality because I couldn't figure it out.